File: problem_6.py
def solution_6_part_1(limit):
    result = 0
    for index in range(1, limit + 1):
        result += index ** 2
    return result


def solution_6_part_2(limit):
    result = 0
    for index in range(1, limit + 1):
        result += index
    return result ** 2


print(True if (solution_6_part_2(100) - solution_6_part_1(100) == 25164150) else False)

File: go/solutions/problem6.go
package solutions

// Solution6 => void -> string
func Solution6() string {
	calcResult := squareOfSum(100) - sumOfSquares(100)
	if calcResult == 25164150 {
		return "problem 6 solved"
	}
	return "problem 6 not solved"
}

func sumOfSquares(limit int) int {
	result := 0
	for index := 1; index <= limit; index++ {
		result += index * index
	}
	return result
}

func squareOfSum(limit int) int {
	result := 0
	for index := 1; index <= limit; index++ {
		result += index
	}
	return result * result
}
